brut.apktool/apktool-lib/src/main/java/brut/androlib/res/data/ResResSpec.java

```diff
@@ -36,11 +36,14 @@ public ResResSpec(ResID id, String name, ResPackage pkg, ResTypeSpec type) {
         this.mId = id;
         String cleanName;
 
+        if (name == null || name.isEmpty() || name.equals("(name removed)"))
+            name = "APKTOOL_DUMMYVAL_" + id.toString();
+
```

Comment:
Could we move this back into the `else` statement? We only need this logic path if the res spec is null.

Reply:
can change for fix by one-liner
```java
        String cleanName;

        name = ("(name removed)".equals(name)) ? null : name;

        ResResSpec resResSpec = type.getResSpecUnsafe(name);
```

Reply:
changed to
```java
name = (("(name removed)".equals(name)) ? null : name);
```